metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala

```diff
@@ -133,8 +140,53 @@ final class FormattingProvider(
     }
   }
 
-  private def runFormat(path: AbsolutePath, input: Input): List[l.TextEdit] = {
-    val fullDocumentRange = Position.Range(input, 0, input.chars.length).toLSP
+  /**
+   * if the configuration of Scalafmt for the project is properly done,
+   * this method is going to produce a List of a single TextEdit for the
+   * formatted string of the whole file, regardless of whether the file is
+   * already well formatted or not.
+   * @param path the path to the file we want to get the formatting [[TextEdit]]s for
+   */
+  def programmaticallyFormat(
+      path: AbsolutePath,
+      token: CancelChecker
+  ): Future[util.List[l.TextEdit]] = {
```

Comment:
Why would you need to return something always irrelevant of whether the file was formatter?

Reply:
This way,  only when the Scalafmt config file is not present, we would receive an empty result.
Also, if the file is not changed, still, we need the result to feed into the rest of the processing for removing the braces.